apps/modernization-ui/src/authorization/IdleTimer.tsx

```diff
@@ -0,0 +1,86 @@
+import { Button } from 'components/button';
+import React, { useState, useEffect } from 'react';
+
+interface IdleTimerProps {
+    timeout: number; // Timeout in milliseconds
+    onIdle: () => void; // Callback function to execute when idle
+}
+
+const IdleTimer: React.FC<IdleTimerProps> = ({ timeout, onIdle }) => {
+    const [idle, setIdle] = useState(false);
+    const [showWarningModal, setShowWarningModal] = useState(false);
+
+    useEffect(() => {
+        let idleTimer: number;
+        let warningTimer: number;
+
+        const resetIdleTimer = () => {
+            clearTimeout(idleTimer);
+            idleTimer = window.setTimeout(() => {
+                setIdle(true);
+                removeEventListeners();
+                startWarningTimer();
+            }, timeout);
+        };
+
+        const startWarningTimer = () => {
+            setShowWarningModal(true);
+            clearTimeout(warningTimer);
+            warningTimer = window.setTimeout(
+                () => {
+                    onIdle();
+                    setShowWarningModal(false);
+                },
+                1000 * 60 * 5
```

Comment:
suggestion:  It would be useful to have the warning duration passed through a property also.